controllers/topology/desired_state.go

```diff
@@ -170,6 +182,106 @@ func computeCluster(current *clusterTopologyState, infrastructureCluster, contro
 	return cluster
 }
 
+// computeMachineDeployment computes the desired state for a MachineDeploymentTopology.
+// The generated machineDeployment object is calculated using the values from the machineDeploymentTopology and
+// the machineDeployment class.
+func computeMachineDeployment(class *clusterTopologyClass, current *clusterTopologyState, topology clusterv1.MachineDeploymentTopology) (machineDeploymentTopologyState, error) {
+	desiredMachineDeployment := machineDeploymentTopologyState{}
+
+	className := topology.Class
+	machineDeploymentClass, err := getMachineDeploymentClass(class, className)
+	if err != nil {
+		return desiredMachineDeployment, err
+	}
+
+	machineDeploymentTopologyClass, ok := class.machineDeployments[className]
+	if !ok {
+		return desiredMachineDeployment, errors.Errorf("class name %s not found in clusterClass %s", className, class.clusterClass.Name)
+	}
+
+	currentMachineDeployment := current.machineDeployments[topology.Name]
+	var currentBootstrapTemplate *corev1.ObjectReference
```

Comment:
nit
```suggestion
	var currentBootstrapTemplateRef *corev1.ObjectReference
```
(same for the infrastructureRef)